ext/opentelemetry-ext-cloud-trace/tests/test_cloud_trace_exporter.py

```diff
@@ -0,0 +1,245 @@
+        self.client_patcher.stop()
+
+    def test_constructor_default(self):
+        exporter = CloudTraceSpanExporter(self.project_id)
+        self.assertEqual(exporter.project_id, self.project_id)
+
+    def test_constructor_explicit(self):
+        client = mock.Mock()
+        exporter = CloudTraceSpanExporter(self.project_id, client=client)
+
+        self.assertIs(exporter.client, client)
+        self.assertEqual(exporter.project_id, self.project_id)
+
+    def test_export(self):
+        trace_id = "6e0c63257de34c92bf9efcd03927272e"
+        span_id = "95bb5edabd45950f"
+        span_datas = [
+            Span(
+                name="span_name",
+                context=SpanContext(
+                    trace_id=int(trace_id, 16),
+                    span_id=int(span_id, 16),
+                    is_remote=False,
+                ),
+                parent=None,
+                kind=SpanKind.INTERNAL,
+            )
+        ]
+
+        cloud_trace_spans = {
+            "name": "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            ),
+            "span_id": span_id,
+            "parent_span_id": None,
+            "display_name": {"value": "span_name", "truncated_byte_count": 0},
+            "attributes": {"attribute_map": {}},
+            "links": None,
+            "status": None,
+            "time_events": None,
+            "start_time": None,
+            "end_time": None,
+        }
+
+        client = mock.Mock()
+
+        exporter = CloudTraceSpanExporter(self.project_id, client=client)
+
+        exporter.export(span_datas)
+
+        client.create_span.assert_called_with(**cloud_trace_spans)
+        self.assertTrue(client.create_span.called)
+
+    def test_extract_status(self):
+        self.assertIsNone(_extract_status(None))
+        self.assertEqual(
+            _extract_status(Status(canonical_code=StatusCanonicalCode.OK)),
+            {"details": None, "code": 0},
+        )
+        self.assertEqual(
+            _extract_status(
+                Status(
+                    canonical_code=StatusCanonicalCode.UNKNOWN,
+                    description="error_desc",
+                )
+            ),
+            {"details": None, "code": 2, "message": "error_desc"},
+        )
+
+    def test_extract_attributes(self):
+        self.assertEqual(_extract_attributes({}), {"attribute_map": {}})
+        self.assertEqual(
+            _extract_attributes(self.attributes_variety_pack),
+            self.extracted_attributes_variety_pack,
+        )
+        # Test ignoring attributes with illegal value type
+        self.assertEqual(
+            _extract_attributes({"illegal_attribute_value": dict()}),
+            {"attribute_map": {}},
+        )
+
+    def test_extract_events(self):
+        self.assertIsNone(_extract_events([]))
+        time_in_ns = 1589919268850900051
+        time_in_ms_and_ns = {"seconds": 1589919268, "nanos": 850899968}
```

Comment:
nit: you should name these `time_in_ns1` and `time_in_ms_and_ns1` for consistency with the `event1` vs `event2` naming.